### PR TITLE
Rename param sql_statement to sql in get_value_list

### PR DESCRIPTION
**Pull request by pankajastro**
# Description
Rename param `sql_statement` to `sql` in `get_value_list` dynamic task template pattern in order to make public interfaces more consistent. 
closes: #865 #866

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Review by pankajkoti — COMMENTED** [review]
There is a `run_sql` method in `databases/base.py` which still accepts `sql_statement` as an argument https://github.com/astronomer/astro-sdk/blob/ddf7f12b723595d88991d286fc39612cec6de455/python-sdk/src/astro/databases/base.py#L82
I think that may still cause a confusion and we might use `sql_statement` across other places later by considering this argument. I believe we called it `sql_statement` here  in `get_value_list` by referring to that in the first place. In my opinion, it makes sense to rename that argument to `sql` too and correct the usages in our codebase.

**Comment by pankajastro** [comment]
> There is a `run_sql` method in `databases/base.py` which still accepts `sql_statement` as an argument
> 
> https://github.com/astronomer/astro-sdk/blob/ddf7f12b723595d88991d286fc39612cec6de455/python-sdk/src/astro/databases/base.py#L82
> 
> . I think that may still cause a confusion and we might use `sql_statement` across other places later by considering this argument. I believe we called it `sql_statement` here by referring to that in the first place. In my opinion, it makes sense to rename that argument to `sql` too and correct the usages in our codebase.

`run_sql` we have already released so not sure whether to deprecate that now or not. WDYT?

**Comment by pankajkoti** [comment]
> > There is a `run_sql` method in `databases/base.py` which still accepts `sql_statement` as an argument
> > https://github.com/astronomer/astro-sdk/blob/ddf7f12b723595d88991d286fc39612cec6de455/python-sdk/src/astro/databases/base.py#L82
> > 
> > . I think that may still cause a confusion and we might use `sql_statement` across other places later by considering this argument. I believe we called it `sql_statement` here by referring to that in the first place. In my opinion, it makes sense to rename that argument to `sql` too and correct the usages in our codebase.
> 
> `run_sql` we have already released so not sure whether to deprecate that now or not. WDYT?

Since this is a base class method, I don't think external users are using it. My point to rename this was so that we have the argument consistent in our codebase and no one later uses `sql_statement` in other places by referring to this argument name.

What are your thoughts? @tatiana @phanikumv @utkarsharma2 @sunank200 @feluelle @kaxil ?

**Comment by kaxil** [comment]
@pankajastro  PR description has some unfilled details !

**Comment by kaxil** [comment]
>Since this is a base class method, I don't think external users are using it. My point to rename this was so that we have the argument consistent in our codebase and no one later uses sql_statement in other places by referring to this argument name.


No strong opinion on that since it is not a user-facing class.

**Review by kaxil — APPROVED** [review]

**Comment by phanikumv** [comment]
> > > There is a `run_sql` method in `databases/base.py` which still accepts `sql_statement` as an argument
> > > https://github.com/astronomer/astro-sdk/blob/ddf7f12b723595d88991d286fc39612cec6de455/python-sdk/src/astro/databases/base.py#L82
> > > 
> > > . I think that may still cause a confusion and we might use `sql_statement` across other places later by considering this argument. I believe we called it `sql_statement` here by referring to that in the first place. In my opinion, it makes sense to rename that argument to `sql` too and correct the usages in our codebase.
> > 
> > 
> > `run_sql` we have already released so not sure whether to deprecate that now or not. WDYT?
> 
> Since this is a base class method, I don't think external users are using it. My point to rename this was so that we have the argument consistent in our codebase and no one later uses `sql_statement` in other places by referring to this argument name.
> 
> What are your thoughts? @tatiana @phanikumv @utkarsharma2 @sunank200 @feluelle @kaxil ?

We need to have consistency across the repo and risk is less since its a base class internal method. So my thought is to rename it there as well

**Comment by feluelle** [comment]
I am all-in for deprecating sql_statement and making it consistent across the codebase. Note that renaming it is a breaking change. So we might want to deprecate this arg by creating a new function and calling the new function from the old function and raising a deprecation warning inbetween.

**Comment by pankajastro** [comment]
> I am all-in for deprecating sql_statement and making it consistent across the codebase. Note that renaming it is a breaking change. So we might want to deprecate this arg by creating a new function and calling the new function from the old function and raising a deprecation warning inbetween.

Any thought on new function name?

**Comment by pankajastro** [comment]
> We need to have consistency across the repo and risk is less since its a base class internal method. So my thought is to rename it there as well

renaming looks like a breaking change since there is not sign that this function is internal.

**Comment by utkarsharma2** [comment]
I don't think we should have breaking changes in a minor release. 

cc: @kaxil

**Comment by kaxil** [comment]
We don't need to create a separate `run_sql_query` method, we can just deprecate the kwarg `sql_statement`

File: python-sdk/src/astro/databases/base.py [hunk 1]
```diff
@@ -81,23 +82,36 @@ def run_sql(
...
+            "Please use  `run_sql_query` instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        return self.run_sql_query(sql=sql_statement, parameters=parameters)
+
+    def run_sql_query(
+        self,
+        sql: str | ClauseElement,
+        parameters: dict | None = None,
```
Comment (kaxil) [inline python-sdk/src/astro/databases/base.py:97]:
```suggestion
        sql: str | ClauseElement,
        parameters: dict | None = None,
```

File: python-sdk/src/astro/databases/base.py [hunk 2]
```diff
@@ -81,23 +82,36 @@ def run_sql(
...
-        :param sql_statement: Contains SQL query to be run against database
+        :param sql: Contains SQL query to be run against database
         :param parameters: Optional parameters to be used to render the query
         """
         if parameters is None:
             parameters = {}
 
-        if isinstance(sql_statement, str):
-            result = self.connection.execute(sqlalchemy.text(sql_statement), parameters)
+        if isinstance(sql, str):
```
Comment (kaxil) [inline python-sdk/src/astro/databases/base.py:111]:
```suggestion

        if "sql_statement" in kwargs:
            warnings.warn(
                "`sql_statement` is deprecated and will be removed in future release"
                "Please use  `sql` param instead.",
                DeprecationWarning,
                stacklevel=2,
            )
        sql = sql or kwargs.get("sql_statement")    
        if isinstance(sql, str):
```

Comment (kaxil) [reply python-sdk/src/astro/databases/base.py:111]: ^^ something like this (untested)

File: python-sdk/src/astro/databases/base.py (hunk 1 above)
Comment (kaxil) [reply python-sdk/src/astro/databases/base.py:97]: We don't need a new method -- we just need to deprecate a kwarg

Comment (kaxil) [reply python-sdk/src/astro/databases/base.py:97]: something like https://github.com/apache/airflow/commit/b220fe60d5302546d85ee15cf6c8eaa5859316d3

File: python-sdk/src/astro/databases/base.py (hunk 2 above)
Comment (utkarsharma2) [reply python-sdk/src/astro/databases/base.py:111]: Then it won't be a breaking change either

**Comment by pankajastro** [comment]
I'm ignoring code coverage at few places in this PR created a followup ticket to fix it https://github.com/astronomer/astro-sdk/issues/870

**Review by utkarsharma2 — APPROVED** [review]

**Pull request merged**